### PR TITLE
Add PDF download to surveys

### PR DESCRIPTION
**Pull request by jonasdeluna**
# Description

Depends on: https://github.com/webkom/lego/pull/3668

Adds PDF download to surveys

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

> [!CAUTION]
> Make sure your images do not contain any real user information.

<table>
    <tr>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td></td>
        <td></td>
    </tr>
</table>

# Testing

- [ x] I have thoroughly tested my changes.

Tested manually

---

Resolves: ABA-1102

### CODE REVIEW

File: app/routes/surveys/components/AdminSideBar.tsx [hunk 1]
```diff
@@ -90,6 +93,27 @@ const AdminSideBar = ({
                 </Button>
               ))}
 
+            {actionGrant &&
+              actionGrant.includes('csv') &&
```
Comment (ivarnakken) [inline app/routes/surveys/components/AdminSideBar.tsx:97]:
```suggestion
            {actionGrant?.includes('csv') &&
```

**Review by ivarnakken — APPROVED** [review]
Looks good! :shipit:

File: app/routes/surveys/components/AdminSideBar.tsx [hunk 2]
```diff
@@ -69,9 +72,8 @@ const AdminSideBar = ({
               Rediger
             </LinkButton>
 
-            {actionGrant &&
-              actionGrant.includes('csv') &&
-              exportSurvey &&
+            {actionGrant.includes('csv') &&
```
Comment (ivarnakken) [inline app/routes/surveys/components/AdminSideBar.tsx:75]: `actionGrant` might be `undefined`, no?
```suggestion
            {actionGrant?.includes('csv') &&
```

File: app/routes/surveys/components/AdminSideBar.tsx [hunk 3]
```diff
@@ -83,13 +85,33 @@ const AdminSideBar = ({
...
                 <Button
-                  onPress={async () => setGeneratedCSV(await exportSurvey())}
+                  onPress={async () => setGeneratedCSV(await exportSurveyCSV())}
                 >
                   <Icon iconNode={<FileUp />} size={19} />
                   Eksporter til CSV
                 </Button>
               ))}
 
+            {actionGrant.includes('csv') &&
```
Comment (ivarnakken) [inline app/routes/surveys/components/AdminSideBar.tsx:95]:
```suggestion
            {actionGrant?.includes('csv') &&
```

**Comment by ch0rizo** [comment]
LGMT!

**Review by ch0rizo — COMMENTED** [review]
LGMT!

**Pull request merged**